examples/type-level-boolean-programming

```diff
@@ -0,0 +1,39 @@
+/// A list of type `a` that uses a Boolean at the type level to trackwhether it has
```

Comment:
Why is this here? :)

Reply:
That's a really good question and I don't know the answer